src/components/wizards/create-vm/CreateVMWizard.js

```diff
@@ -0,0 +1,366 @@
+              return null;
+            }
+            let child = null;
+            const valid = _.get(this.state.basicSettings[key], 'valid', null);
+            if (this.basicSettings[key].type === 'text') {
+              child = <FormControl type="text" value={_.get(this.state.basicSettings[key], 'value', '')} onChange={event => this.onFormChange(event.target.value, key)} />;
+            }
+            if (this.basicSettings[key].type === 'textarea'){
+              child = <FormControl componentClass="textarea" value={_.get(this.state.basicSettings[key], 'value', '')} onChange={event => this.onFormChange(event.target.value, key)} />;
+            }
+            if (this.basicSettings[key].type === 'dropdown'){
+              child = <ButtonGroup justified>
+                <DropdownButton
+                  id={`wizard-dropdown-${key}`}
+                  bsStyle="default"
+                  className="wizard-dropdown"
+                  title={_.has(this.state.basicSettings[key], 'value') ? this.state.basicSettings[key].value : this.basicSettings[key].default}
+                  onSelect={value => this.onFormChange(value, key)}>
+                  {this.basicSettings[key].values.map(value => <MenuItem key={value.name} eventKey={value.value ? value.value : value.name}>{value.name}</MenuItem>)}
+                </DropdownButton>
+              </ButtonGroup>;
+            }
+            if (this.basicSettings[key].type === 'checkbox') {
+              child = <Checkbox checked={_.get(this.state.basicSettings[key], 'value', false)} onChange={event => this.onFormChange(event.target.checked, key)}>{this.basicSettings[key].title}</Checkbox>;
+            }
+            return <FormGroup key={key} validationState={valid ? 'error' : null}>
+              <Col sm={_.get(this.basicSettings[key], 'labelSize') ? this.basicSettings[key].labelSize : 3} className="text-right">
+                {this.basicSettings[key].type === 'checkbox' ? null:
+                  <ControlLabel className={this.basicSettings[key].required ? 'required-pf': null}>{this.basicSettings[key].title}</ControlLabel>}
+              </Col>
+              <Col sm={_.get(this.basicSettings[key], 'labelSize') ? 8-this.basicSettings[key].labelSize : 5}>
+                {child}
+                <HelpBlock>{valid ? valid: null}</HelpBlock>
+              </Col>
+            </FormGroup>;
+          })}
+        </Form>
+      },
+      {
+        title: 'Networking',
+        render: () => <p>Render</p>
+      },
+      {
+        title: 'Storage',
+        render: () => <StorageTable storageClasses={this.props.storageClasses}/>
+      },
+      {
+        title: 'Result',
+        render: () => {
+          this.props.k8sCreate(this.createVM(this.state.basicSettings, this.state.network, this.state.storage))
+          return <p>object visible in console</p>
+        }
+      }
+    ]
+
+    createVM = (basicSettings, network, storage) => {
```

Comment:
I would rather see this in another file and give it a data in stable format. Maybe some file like `requests.js` ? Then it would create final data structure vm.

Reply:
OK, I moved it to k8s/request.js